climpred/classes.py

```diff
@@ -228,20 +416,38 @@ def generate_uninitialized(self, var=None):
         Returns:
             Bootstrapped (uninitialized) ensemble as a Dataset.
         """
+        has_dataset(
+            self._datasets['control'], 'control', 'generate an uninitialized ensemble.'
+        )
+
         if var is not None:
             uninit = bootstrap_uninit_pm_ensemble_from_control(
-                self.initialized[var], self.control[var]
+                self._datasets['initialized'][var], self._datasets['control'][var]
             ).to_dataset()
         else:
             uninit = bootstrap_uninit_pm_ensemble_from_control(
-                self.initialized, self.control
+                self._datasets['initialized'], self._datasets['control']
             )
-        self.uninitialized = uninit
-
-    def compute_metric(self, metric='pearson_r', comparison='m2m'):
+        datasets = self._datasets.copy()
+        datasets.update({'uninitialized': uninit})
+        return self._construct_direct(datasets)
+
+    # -----------------
+    # Getters & Setters
+    # -----------------
+    def get_control(self):
```

Comment:
https://www.geeksforgeeks.org/getter-and-setter-in-python/
I think we may want to use `@property` decorator, but I don't really know how to write classes very well myself.